pkg/api/validation/validation.go

```diff
@@ -584,6 +585,7 @@ func ValidatePodUpdate(newPod, oldPod *api.Pod) errs.ValidationErrorList {
 		allErrs = append(allErrs, errs.NewFieldInvalid("spec.containers", newPod.Spec.Containers, "some fields are immutable"))
 	}
 
+	newPod.Status = oldPod.Status
```

Comment:
This probably could be moved into 4248 - it's still correct, but 4248 is where the reset really needs to happen.